# Runnable的缺陷

![](image/Pasted%20image%2020220206193710.png)

```java
package future;

import java.util.concurrent.Callable;
import java.util.concurrent.Future;

/**
 * 描述：     在run方法中无法抛出checked Exception
 */
public class RunnableCantThrowsException {

    public void ddd() throws Exception {
    }

    public static void main(String[] args) {
        Runnable runnable = new Runnable() {
            @Override
            public void run() {
                try {
                    throw new Exception();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        };


    }
}

```

# Callable接口

# Future类

## 线程池submit方法返回Future对象

## 用FutureTask创建Future

## Future的注意点

